### DISCUSSION TITLE
Active tower damping - side-side vibrations

### DISCUSSION THREAD

**Discussion by roger1017** [NatLabRockies/ROSCO#187] [General]
Hi ROSCO development team,

I would like to know if ROSCO has the feature to improve side side tower vibration by actuating electric torque?

Many thanks!

BR
Roger

**Comment by dzalkind** [comment]
Hi Roger, it's mid-way through development in #164.  Feel free to try it out.

The current issue with it is that the torque gains were not high enough to adequately change the rotor speed.  We'd ideally like to smoothly transition the gains to higher values when the set points are changed.  If you're interested in continuing the development, let me know if you have any questions.

**Reply by GertHeirman** [reply]
Dear @dzalkind,

I had a look at the source code in #164. It seems to me that this functionality focuses on making sure the controller does not excite the turbine in critical frequency ranges. I can image a generator torque fluctuation at the wrong frequency would lead to excessive side-side motion of the tower.

Is there any feature to actively dampen side-side tower vibration, e.g. actively adding/subtracting an additional generator torque based on a controller which uses e.g. side-side nacelle acceleration as input?
